src/hw/connectApp.js

```diff
@@ -160,19 +167,53 @@ const cmd = ({
   appName,
   requiresDerivation,
   dependencies,
+  requireLatestFirmware,
 }: Input): Observable<ConnectAppEvent> =>
   withDevice(devicePath)((transport) =>
     Observable.create((o) => {
       const timeoutSub = of({ type: "unresponsiveDevice" })
         .pipe(delay(1000))
         .subscribe((e) => o.next(e));
 
-      const innerSub = ({ appName, dependencies }: any) =>
+      const innerSub = ({
+        appName,
+        dependencies,
+        requireLatestFirmware,
+      }: any) =>
         defer(() => from(getAppAndVersion(transport))).pipe(
           concatMap((appAndVersion): Observable<ConnectAppEvent> => {
             timeoutSub.unsubscribe();
 
             if (isDashboardName(appAndVersion.name)) {
+              // check if we meet minimum fw
+              if (requireLatestFirmware) {
+                return from(getDeviceInfo(transport)).pipe(
+                  mergeMap((deviceInfo: DeviceInfo) =>
+                    from(manager.getLatestFirmwareForDevice(deviceInfo)).pipe(
+                      mergeMap((latest: ?FirmwareUpdateContext) => {
+                        if (
+                          !latest || // Fixme is it safe to let it pass here?
```

Comment:
you can remove this comment => yes it is, it means that it's up to date.

Reply:
Done